playbooks/pki.yml

```diff
@@ -173,7 +174,7 @@
           no_log: true
           become: true
           community.crypto.openssl_privatekey:
-            path: /etc/pki/{{ inventory_hostname }}_{{ cert_key_type | default('') }}.key
+            path: /etc/pki/private/{{ inventory_hostname | basename }}_{{ cert_key_type | default('') }}.key
```

Comment:
Could you please let me know if you would be fine with this alternative line:
```yml
        path: "/etc/pki/private/{{ inventory_hostname | basename }}{% if cert_key_type %}_{{ cert_key_type }}{% endif %}.key"
```

Currently, if `cert_key_type` isn't defined, then the key name will end with `_.key`, which isn't very pretty. What do you think about removing this underscore in case no key type is provided?

Reply:
I'd rather add a task var to make sure the `type` module param is always set.

Reply:
So your approach would be to use the following instead:
```yml
          community.crypto.openssl_privatekey:
            path: /etc/pki/private/{{ inventory_hostname | basename }}_{{ cert_key_type | default('RSA') }}.key
            mode: 0600
            size: "{{ cert_key_size | default(omit) }}"
            type: "{{ cert_key_type | default('RSA') }}"
            return_content: true
```

Isn't that more dangerous that anything else to force a `type` to be present and not omitted by default? If we force a default to "RSA"(or something else), then it might break at some point in the future, if this type is removed.

Reply:
Yes that's what I had in mind except I would have used a task level var to a

```yaml
- name: ....
  vars:
    key_type: "{{ cert_key_type | default('RSA') }}"
  community.crypto.openssl_privatekey:
    path: /etc/pki/private/{{ inventory_hostname | basename }}_{{ key_type }}.key
    mode: 0600
    size: "{{ cert_key_size | default(omit) }}"
    type: "{{ key_type }}"
    return_content: true
```

Reply:
It was added